server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnable.java

```diff
@@ -18,42 +18,49 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-package org.sonar.server.computation;
+package org.sonar.server.computation.queue;
 
 import com.google.common.base.Optional;
 import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
 import org.sonar.core.util.logs.Profiler;
 import org.sonar.db.ce.CeActivityDto;
+import org.sonar.server.computation.queue.report.ReportTaskProcessor;
 
 import static java.lang.String.format;
 
-public class CeWorkerImpl implements CeWorker {
+class CeWorkerRunnable implements Runnable {
 
-  private static final Logger LOG = Loggers.get(CeWorkerImpl.class);
+  private static final Logger LOG = Loggers.get(CeWorkerRunnable.class);
 
   private final CeQueue queue;
   private final ReportTaskProcessor reportTaskProcessor;
 
-  public CeWorkerImpl(CeQueue queue, ReportTaskProcessor reportTaskProcessor) {
+  public CeWorkerRunnable(CeQueue queue, ReportTaskProcessor reportTaskProcessor) {
     this.queue = queue;
     this.reportTaskProcessor = reportTaskProcessor;
   }
 
   @Override
   public void run() {
-    CeTask task;
+    Optional<CeTask> ceTask = tryAndFindTaskToExecute();
+    if (!ceTask.isPresent()) {
+      return;
+    }
+
+    executeTask(ceTask.get());
+  }
+
+  private Optional<CeTask> tryAndFindTaskToExecute() {
     try {
-      Optional<CeTask> taskOpt = queue.peek();
-      if (!taskOpt.isPresent()) {
-        return;
-      }
-      task = taskOpt.get();
+      return queue.peek();
     } catch (Exception e) {
       LOG.error("Failed to pop the queue of analysis reports", e);
-      return;
     }
+    return Optional.absent();
+  }
 
+  private void executeTask(CeTask task) {
     // TODO delegate the message to the related task processor, according to task type
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)